ttnn/cpp/ttnn/distributed/distributed_pybind.cpp

```diff
@@ -159,6 +159,23 @@ void py_module(py::module& module) {
                     MeshSubDeviceManagerId: The ID of the created sub-device manager.
             )doc")
         .def(
+            "create_sub_device_manager",
+            [](MeshDevice& self,
+               const std::vector<std::vector<SubDevice>>& mesh_sub_devices,
+               DeviceAddr local_l1_size) { return self.create_sub_device_manager(mesh_sub_devices, local_l1_size); },
```

Comment:
For single device, the create_sub_device_manager api takes a list of SubDevices to configure the device with.
The original create_sub_device_manager api for MeshDevice also only takes a single list of SubDevices, and that is used to configure all devices in the mesh so each device has the same sub-device configuration.
This api takes in a list of list so that user can specify a different sub-device configuration per device in the mesh_device. This is mainly required because ethernet cores are not the same across chips.
You're suggesting to only have the api that takes a single list, and to check the size at runtime? ie if it's the same size as the number of mesh devices then use it for all devices, if not then assert it's that the list size is a multiple of num devices and split accordingly? This doesn't seem as clear to me as a user compared to having separate overloads.

Reply:
Mostly asking for context - thanks! Let's update the documentation in python mentioning that every element in the list is for a different device?

(IIUC the alternative approach of getting individual devices and setting up subdevices on them doesn't work, because we need a single subdevice manager per mesh?)

Reply:
Right now we have a subdevice manager per device, and the cores could be different even though it's really the same target configuration due to the eth cores being different per device.

The new distributed proposal will have 1 manager per mesh, and eth cores will no longer be part of the sub-devices because it will be part of fabric, so we wouldn't have this issue anymore.

Reply:
I've added the issue number + updated the documentation.